fdbserver/include/fdbserver/IPageEncryptionKeyProvider.actor.h

```diff
@@ -341,6 +335,10 @@ class TenantAwareEncryptionKeyProvider : public IPageEncryptionKeyProvider {
 		if (key.startsWith(systemKeysPrefix)) {
 			return { SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID, systemKeysPrefix.size() };
 		}
+		// Cluster-aware encryption.
+		if (encryptionMode == EncryptionAtRestMode::CLUSTER_AWARE) {
+			return { FDB_DEFAULT_ENCRYPT_DOMAIN_ID, 0 };
+		}
```

Comment:
For extracting the tenant id (lines 356-347) we have a helper method TenantAPI::extractTenantIdFromKeyRef, we should probably use that

Reply:
Yeah, let's reuse those method, I remember making Nim work extra to make it a generic method, only for this day :D .